src/providers/joystream/provider.tsx

```diff
@@ -1,77 +1,100 @@
 import { web3FromAddress } from '@polkadot/extension-dapp'
-import React, { useCallback, useEffect, useState } from 'react'
+import { ProxyMarked, Remote, proxy, wrap } from 'comlink'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
 
 import { NODE_URL } from '@/config/urls'
 import { JoystreamJs } from '@/joystream-lib'
 import { useEnvironmentStore } from '@/providers/environment/store'
 import { SentryLogger } from '@/utils/logs'
+import JoystreamJsWorker from '@/web-workers/polkadot?worker'
 
 import { useConnectionStatusStore } from '../connectionStatus'
 import { useUser } from '../user'
 
 type JoystreamContextValue = {
-  joystream: JoystreamJs | null
+  joystream: Remote<JoystreamJs> | undefined
+  proxyCallback: <T extends object>(callback: T) => T & ProxyMarked
 }
 export const JoystreamContext = React.createContext<JoystreamContextValue | undefined>(undefined)
 JoystreamContext.displayName = 'JoystreamContext'
+const worker = new JoystreamJsWorker()
+const api = wrap<typeof JoystreamJs>(worker)
 
 export const JoystreamProvider: React.FC = ({ children }) => {
   const { activeAccountId, accounts } = useUser()
   const { nodeOverride } = useEnvironmentStore((state) => state)
   const setNodeConnection = useConnectionStatusStore((state) => state.actions.setNodeConnection)
-
-  const [joystream, setJoystream] = useState<JoystreamJs | null>(null)
-
+  const [initialized, setInitialized] = useState(false)
   const handleNodeConnectionUpdate = useCallback(
     (connected: boolean) => {
       setNodeConnection(connected ? 'connected' : 'disconnected')
     },
     [setNodeConnection]
   )
+  const joystream = useRef<Remote<JoystreamJs> | undefined>()
 
-  useEffect(() => {
-    let joystream: JoystreamJs
+  const proxyCallback = <T extends object>(callback: T) => proxy(callback)
 
-    const init = async () => {
+  useEffect(() => {
+    const getjoystream = async () => {
       try {
         setNodeConnection('connecting')
-        joystream = new JoystreamJs(nodeOverride || NODE_URL)
-        setJoystream(joystream)
-
-        joystream.onNodeConnectionUpdate = handleNodeConnectionUpdate
+        joystream.current = await new api(nodeOverride ?? NODE_URL, proxy(handleNodeConnectionUpdate))
+        setInitialized(true)
       } catch (e) {
         handleNodeConnectionUpdate(false)
-        SentryLogger.error('Failed to create JoystreamJS instance', 'JoystreamProvider', e)
+        SentryLogger.error('Failed to create JoystreamJS joystream', 'JoystreamProvider', e)
       }
     }
-
-    init()
-
+    getjoystream()
     return () => {
-      joystream?.destroy()
+      const destroy = async () => {
+        if (joystream.current) {
+          await joystream.current.destroy()
+        }
+      }
+      destroy()
```

Comment:
What's a difference between this and:

```suggestion
      joystream.current?.destroy()
```